hologram_dashboard/fetcher/hologram_data_source_availability_fetcher.cc

```diff
@@ -63,17 +63,57 @@ void HologramDataSourceAvailabilityFetcher::GetHologramDataAvailability(
     // }
 }
 
-void HologramDataSourceAvailabilityFetcher::UpdateDataAvailability(
-    std::string system, absl::Time date, DataSource data_source, 
-    StatusType status){
-    // TODO(alexanderlin): Implement.
-    // Call UpdateHistory for updating proto's history
+void HologramDataSourceAvailabilityFetcher::UpdateDataAvailability(System system, 
+    absl::Time time, DataSource data_source, StatusType status) {
+    absl::flat_hash_map<DataSource,HologramDataAvailability>& 
+        data_to_availability_map = system_to_data_source_availability_[system];
+    HologramDataAvailability& availability_proto = 
+        data_to_availability_map[data_source];
+    DataSourceDetail* latest_status = availability_proto.mutable_latest_status();
+    if (!availability_proto.has_data_source()) {
+        availability_proto.set_data_source(data_source);
+    }
+    latest_status->set_date(absl::ToUnixSeconds(time));
```

Comment:
I think maybe we should use string to indicate the date for better readability, like "2020-06-16". 

Can you update the proto?

Reply:
My thought is user should never read the raw data, and that using ints instead makes it so that we can be more accurate programmically. By the time user reads the value it would've been translated to a human readable format.

Reply:
using int64 for date makes the proto unreadable.

In our use case, for a DataSource, we only need one DataSourceDetail in a single day.

Reply:
should I make the history in string but latest status in int64? The reason for that is I want to maintain high level of precision for when I want to print to the user when was the last time we checked for done files. That way the user can see that it refreshed 15 minutes ago and is most likely most updated instead of kept in the dark on when was the last time it ran.

Reply:
Try this

message DataSourceDetail {
    optional string date = 1;
    optional StatusType status = 2;
    // The timestamp when this DataSourceDetails is fetched.
    optional int64 timestamp_millis = 3;
}

Reply:
Done